guided_diffusion/models.py

```diff
@@ -191,8 +192,35 @@ def forward(self, x):
         return x+h_
 
 
-class Model(nn.Module, PyTorchModelHubMixin):
-    def __init__(self, config):
+def namespace2dict(config: Namespace) -> dict: 
+  v = dict(vars(config))
+  for key,value in v.items():
+    if isinstance(value, Namespace):
+      v[key] = namespace2dict(value)
+  return v
+
+
+def dict2namespace(config: dict) -> Namespace:
```

Comment:
(nit) method is already defined in `./utils`. Might be worth moving `namespace2dict` as well + import them